src/ansys/motorcad/core/geometry.py

```diff
@@ -333,19 +392,87 @@ def is_closed(self):
             Whether region is closed
         """
         if len(self._entities) > 0:
-            entity_first = self._entities[0]
-            entity_last = self._entities[-1]
+            # Make sure all adjacent entities have a point in common
+            return all(
+                get_entities_have_common_coordinate(self._entities[i - 1], self._entities[i])
+                for i in range(0, len(self._entities))
+            )
+        else:
+            return False
 
-            is_closed = get_entities_have_common_coordinate(entity_first, entity_last)
+    def is_nonintersecting(self):
+        """Check whether region entities intersect each other.
 
-            for i in range(len(self._entities) - 1):
-                is_closed = get_entities_have_common_coordinate(
-                    self._entities[i], self._entities[i + 1]
-                )
+        Returns
+        -------
+        Boolean
+            Whether region is nonintersecting
+        """
+        side_pairs = []
```

Comment:
I prefer entity pairs over side_pairs
```suggestion
        entity_pairs = []
```